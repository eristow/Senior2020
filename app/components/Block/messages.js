/*
 * Block Messages
 *
 * This contains all the text for the Block component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.Block';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the Block component!',
  },
});
