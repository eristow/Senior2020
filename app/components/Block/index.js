/**
 *
 * Block
 *
 */

import React from 'react';
// import PropTypes from 'prop-types';
// import styled from 'styled-components';

import { FormattedMessage } from 'react-intl';
import messages from './messages';

function Block() {
  return (
    <div>
      <FormattedMessage {...messages.header} />
    </div>
  );
}

Block.propTypes = {};

export default Block;
