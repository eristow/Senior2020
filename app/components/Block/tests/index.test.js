/**
 *
 * Tests for Block
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Block from '../index';

describe('<Block />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Block />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Block />);
    expect(firstChild).toMatchSnapshot();
  });
});
