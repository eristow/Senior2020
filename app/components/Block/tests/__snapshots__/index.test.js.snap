// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Block /> Should render and match the snapshot 1`] = `
<div>
  <span>
    This is the Block component!
  </span>
</div>
`;
